specification/machinelearningservices/resource-manager/Microsoft.MachineLearningServices/preview/2024-01-01-preview/workspaceRP.json

```diff
@@ -1051,15 +1044,41 @@
           {
             "$ref": "#/parameters/WorkspaceNameParameter"
           },
+          {
+            "$ref": "#/parameters/EndpointNameParameter"
+          },
           {
             "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Endpoint resource object",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/EndpointResourcePropertiesBasicResource"
+            }
           }
         ],
         "responses": {
           "200": {
-            "description": "Success",
+            "description": "The request was successful; the request was well-formed and received properly.",
             "schema": {
-              "$ref": "#/definitions/ExternalFQDNResponse"
+              "$ref": "#/definitions/EndpointResourcePropertiesBasicResource"
+            }
+          },
+          "202": {
```

Comment:
>202 [](http://example.com/codeflow?start=11&length=3)

Async status response for PUT + provisioningState is 201.

Reply:
This is caused by the whole service's structure, we return 202 with location header from the start and all the service calling us are expecting this pattern, change to 201 will need to require a major refactor which currently we don't have bandwidth todo yet.

Reply:
How about just special casing this one (and any other new ones) to do it right, and doing the major refactor later?

Reply:
I totally agreed on this but at this time we already have many service calling this API and we are on a tight schedule to ship this so hopefully we can get an exemption this time, will make sure new API in the future a following the 201 pattern.

Reply:
@rkmanda Would you like to approve or deny this request for exemption?